pkg/controllers/vrep/replication_reconciler.go

```diff
@@ -0,0 +1,311 @@
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package vrep
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	vapi "github.com/vertica/vertica-kubernetes/api/v1"
+	v1beta1 "github.com/vertica/vertica-kubernetes/api/v1beta1"
+	"github.com/vertica/vertica-kubernetes/pkg/cmds"
+	"github.com/vertica/vertica-kubernetes/pkg/controllers"
+	vdbcontroller "github.com/vertica/vertica-kubernetes/pkg/controllers/vdb"
+	verrors "github.com/vertica/vertica-kubernetes/pkg/errors"
+	"github.com/vertica/vertica-kubernetes/pkg/events"
+	vmeta "github.com/vertica/vertica-kubernetes/pkg/meta"
+	"github.com/vertica/vertica-kubernetes/pkg/names"
+	"github.com/vertica/vertica-kubernetes/pkg/vadmin"
+	"github.com/vertica/vertica-kubernetes/pkg/vadmin/opts/replicationstart"
+	"github.com/vertica/vertica-kubernetes/pkg/vk8s"
+	vrepstatus "github.com/vertica/vertica-kubernetes/pkg/vrepstatus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	stateReplicating          = "Replicating"
+	stateSucceededReplication = "Replication successful"
+	stateFailedReplication    = "Replication failed"
+)
+
+type ReplicationReconciler struct {
+	client.Client
+	VRec           *VerticaReplicatorReconciler
+	Vrep           *v1beta1.VerticaReplicator
+	dispatcher     vadmin.Dispatcher
+	SourcePFacts   *vdbcontroller.PodFacts
+	TargetPFacts   *vdbcontroller.PodFacts
+	Log            logr.Logger
+	SourceVdb      *vapi.VerticaDB
+	SourceIP       string
+	TargetVdb      *vapi.VerticaDB
+	TargetIP       string
+	TargetDBName   string
+	SourceUsername string
+	SourcePassword string
+	TargetUsername string
+	TargetPassword string
+}
+
+func MakeReplicationReconciler(cli client.Client, r *VerticaReplicatorReconciler, vrep *v1beta1.VerticaReplicator,
+	log logr.Logger) controllers.ReconcileActor {
+	return &ReplicationReconciler{
+		Client: cli,
+		VRec:   r,
+		Vrep:   vrep,
+		Log:    log.WithName("ReplicationReconciler"),
+	}
+}
+
+func (r *ReplicationReconciler) Reconcile(ctx context.Context, _ *ctrl.Request) (res ctrl.Result, err error) {
+	// no-op if ReplicationComplete is present (either true or false)
+	isPresent := r.Vrep.IsStatusConditionPresent(v1beta1.ReplicationComplete)
+	if isPresent {
+		return ctrl.Result{}, nil
+	}
+
+	// no-op if ReplicationReady is false
+	isFalse := r.Vrep.IsStatusConditionFalse(v1beta1.ReplicationReady)
+	if isFalse {
+		return ctrl.Result{}, nil
+	}
+
+	// fetch both source and target VerticaDBs
+	if res, fetchErr := r.fetchVdbs(ctx); verrors.IsReconcileAborted(res, fetchErr) {
+		return res, fetchErr
+	}
+
+	// determine db names
+	r.determineDBNames()
+
+	// determine usernames and passwords
```

Comment:
your function handles both source and targetDB name, Could we update to `determine usernames and passwords for source and targetDB`?